src/network/networkd-manager.c

```diff
@@ -98,7 +98,7 @@ static int on_connected(sd_bus_message *message, void *userdata, sd_bus_error *r
 
         /* Did we get a timezone or transient hostname from DHCP while D-Bus wasn't up yet? */
         if (m->dynamic_hostname)
-                (void) manager_set_hostname(m, m->dynamic_hostname);
+                (void) manager_set_hostname(m, m->dynamic_hostname, false);
```

Comment:
If `bus` is not ready when setting hostname at first, then, this always try to set transient hostname...

How about the following?
```
# networkd-manager.h
Manager {
    (...);
    char *transient_hostname;
    char *static_hostname;
    (...);
}

# networkd-manager.c
if (m->static_hostname)
    (void) manager_set_hostname(m, m->static_hostname, true);
else if (m->transient_hostname)
    (void) manager_set_hostname(m, m->transient_hostname, false);
```